gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -1009,14 +1065,54 @@ def is_append_contiguous(self, original_dataset: xr.Dataset, append_times: list[
         bool
             whether the original and appending datasets form a contiguous time axis
         """
-        # Use time step of the original dataset to get expected time delta. This could be improved by using Attributes.temporal_resolution
-        # if it were edited to return a numpy.timedelta64 (what if previous dataset has only one time step, for example).
+        # Check if first time in append times is one time step ahead of the last time in the original dataset
         last_time_in_original = original_dataset[self.time_dim].values[-1]
-        expected_delta = last_time_in_original - original_dataset[self.time_dim].values[-2]
+        self.is_time_contiguous(append_times[0], last_time_in_original)
 
-        # Check if first time in append times is one time step ahead of the last time in the original dataset
-        if append_times[0] - last_time_in_original != expected_delta:
-            return False
+        # Check if all times to be appended are contiguous with the prior time step in the append time range
+        self.are_times_contiguous(append_times)
+
+    def are_times_contiguous(self, times: tuple[datetime.datetime]) -> bool:
+        """
+        Convenience method to run `is_time_contiguous` in a loop over an ascending ordered list of input times,
+        since this is a regular pattern
+
+        Parameters
+        ----------
+        times
+            A datetime.datetime object representing the timestamp being checked
+
+        Returns
+        -------
+        bool | None
+            Returns False for any unacceptable timestamp
+        """
+        previous_time = times[0]
+        for time in times[1:]:
+            if self.is_time_contiguous(time, previous_time) is not None:
+                return False
+            previous_time = time
+        return True
+
+
+    def is_time_contiguous(self, time: datetime.datetime, previous_time: datetime.datetime) -> bool:
+        """
+        Return false if a given time is out of order and/or does not follow the previous time, or falls outside of an acceptable range of timedeltas
 
-        # Check if all times to be appended are contiguous with the correct time step
-        return self.are_times_contiguous(append_times, expected_delta)
+        Parameters
+        ----------
+        time
+            A datetime.datetime object representing the timestamp being checked
+        previous_time
+            A datetime.datetime object representing the prior timestamp
+
+        Returns
+        -------
+        bool | None
+            Returns False for any unacceptable timestamp
+        """
+        if self.irregular_update_cadence():
+            if not self.irregular_update_cadence()[0] <= (time - previous_time) <= self.irregular_update_cadence()[1]:
+                return False
+        elif time - previous_time != self.span_to_timedelta():
+            return False
```

Comment:
after this line is where it could return True

Reply:
If it returns true then `is_append_contiguous` will stop there instead of continuing onwards to the `are_times_contiguous` check. 

I think it’s better to just let it return None if nothing bad is encountered.